api/endpoints/slack/interactivity.js

```diff
@@ -0,0 +1,64 @@
+import fetch from 'node-fetch'
+import ensureSlackAuthenticated from '../../ensure-slack-authenticated.js'
+import slashZInner from '../../slash-z-inner.js'
+import Prisma from '../../prisma.js'
+
+export default async (req, res) => {
+  return await ensureSlackAuthenticated(req, res, async () => {
+    // Acknowledge we got the message so Slack doesn't show an error to the user
+    res.status(200).send()
+
+    const payload = JSON.parse(req.body.payload)
+    if (payload.actions.length < 1) return
+
+    switch (payload.actions[0].value) {
+      case 'slash-z-start':
+        // Run /z as usual
+        await slashZInner({
+          displayName: payload.user.name,
+          userId: payload.user.id,
+          channelId: payload.container.channel_id,
+          responseUrl: payload.response_url,
+          skipWarning: true
+        })
+        break
+
+      case 'slash-z-cancel':
+        // Do nothing except delete the ephemeral message
+        await fetch(payload.response_url, {
+          method: 'post',
+          headers: {
+            Authorization: `Bearer ${process.env.SLACK_BOT_USER_OAUTH_ACCESS_TOKEN}`,
+            'Content-Type': 'application/json'
+          },
+          body: JSON.stringify({
+            response_type: 'ephemeral',
+            text: '',
+            delete_original: true
+          })
+        })
+        break
+
+      case 'slash-z-start-ignore':
+        // Run /z and remember not to warn the user again
+        if (!(await Prisma.get('ignoredWarningUser', payload.user.id)).length) {
+          await Prisma.create('ignoredWarningUser', {
+            id: payload.user.id
+          })
+        }
+        await slashZInner({
```

Comment:
Want to move this above the `Prisma.create` line? It's milliseconds, but if someone is waiting on their zoom meeting to start it could help!

Reply:
I kept it in this order to eliminate a tiny bit of confusion in case someone runs /z twice... but now that I think about it, that doesn't really matter either way. Good point, I switched them around.